src/main.cc

```diff
@@ -404,308 +408,316 @@ usage(void)
+        opt_reload_hit_only = 1;
+        break;
 
 #if USE_WIN32_SERVICE
 
-        case 'O':
-            /** \par O
-             * Set global option. opt_command_lin and WIN32_Command_Line */
-            opt_command_line = 1;
-            WIN32_Command_Line = xstrdup(optarg);
-            break;
-#endif
-
-        case 'R':
-            /** \par R
-             * Unset/disable global option opt_reuseaddr */
-            opt_reuseaddr = 0;
-            break;
-
-        case 'S':
-            /** \par S
-             * Set global option opt_store_doublecheck */
-            opt_store_doublecheck = 1;
-            break;
-
-        case 'X':
-            /** \par X
-             * Force full debugging */
-            Debug::parseOptions("rotate=0 ALL,9");
-            Debug::override_X = 1;
-            sigusr2_handle(SIGUSR2);
-            break;
-
-        case 'Y':
-            /** \par Y
-             * Set global option opt_reload_hit_only */
-            opt_reload_hit_only = 1;
-            break;
+    case 'i':
+        /** \par i
+         * Set global option opt_install_service (to TRUE) */
+        opt_install_service = TRUE;
+        break;
+#endif
 
-#if USE_WIN32_SERVICE
+    case 'a':
+    {
+        /** \par a
+         * Add optional HTTP port as given following the option */
+        char *port = xstrdup(optValue);
+        add_http_port(port);
+        xfree(port);
+        break;
+    }
 
-        case 'i':
-            /** \par i
-             * Set global option opt_install_service (to TRUE) */
-            opt_install_service = TRUE;
-            break;
-#endif
-
-        case 'a':
-            /** \par a
-             * Add optional HTTP port as given following the option */
-            add_http_port(optarg);
-            break;
-
-        case 'd':
-            /** \par d
-             * Set global option Debug::log_stderr to the number given following the option */
-            Debug::log_stderr = atoi(optarg);
-            break;
-
-        case 'f':
-            /** \par f
-             * Load the file given instead of the default squid.conf. */
-            xfree(ConfigFile);
-            ConfigFile = xstrdup(optarg);
-            break;
-
-        case 'k':
-            /** \par k
-             * Run the administrative action given following the option */
-
-            /** \li When it is missing or an unknown option display the usage help. */
-            if (!optarg || strlen(optarg) < 1)
-                usage();
-
-            else if (!strncmp(optarg, "reconfigure", strlen(optarg)))
-                /** \li On reconfigure send SIGHUP. */
-                opt_send_signal = SIGHUP;
-            else if (!strncmp(optarg, "rotate", strlen(optarg)))
-                /** \li On rotate send SIGQUIT or SIGUSR1. */
+    case 'd':
+        /** \par d
+         * Set global option Debug::log_stderr to the number given following the option */
+        Debug::log_stderr = atoi(optValue);
```

Comment:
please take the opportunity to fix this old bug. This should be using xatoi() instead of atoi() since we cannot trust the length of the input string to be sane and are able to abort.